spt/OrangeBox/modules/vphysicsDLL.cpp

```diff
@@ -0,0 +1,64 @@
+#include "stdafx.h"
+
+#include "vphysicsDLL.hpp"
+
+#include "convar.h"
+
+#include "..\modules.hpp"
+#include "..\patterns.hpp"
+
+#define DEF_FUTURE(name) auto f##name = FindAsync(ORIG_##name, patterns::inputsystem::##name);
+#define GET_HOOKEDFUTURE(future_name) \
+	{ \
+		auto pattern = f##future_name.get(); \
+		if (ORIG_##future_name) \
+		{ \
+			DevMsg("[vphysics dll] Found " #future_name " at %p (using the %s pattern).\n", \
+			       ORIG_##future_name, \
+			       pattern->name()); \
+			patternContainer.AddHook(HOOKED_##future_name, (PVOID*)&ORIG_##future_name); \
+			for (int i = 0; true; ++i) \
+			{ \
+				if (patterns::vphysics::##future_name.at(i).name() == pattern->name()) \
+				{ \
+					patternContainer.AddIndex((PVOID*)&ORIG_##future_name, i, pattern->name()); \
+					break; \
+				} \
+			} \
+		} \
+		else \
+		{ \
+			DevWarning("[vphysics dll] Could not find " #future_name ".\n"); \
+		} \
+	}
+
+void VPhysicsDLL::Hook(const std::wstring& moduleName,
+                       void* moduleHandle,
+                       void* moduleBase,
+                       size_t moduleLength,
+                       bool needToIntercept)
+{
+	Clear();
+	m_Name = moduleName;
+	m_Base = moduleBase;
+	m_Length = moduleLength;
+	patternContainer.Init(moduleName);
+
+#ifdef SSDK2007
+	this->isgFlagPtr = (bool*)moduleBase + 0xe04d4;
```

Comment:
No pattern, just a hardcoded offset? I guess if there's no hooking then it can only crash on read or write so not a very big deal